lib/invalidateCache.js

```diff
@@ -1,12 +1,46 @@
 var assertExists = require('./assertExists');
 var invalidateModifiedFiles = require('./invalidateModifiedFiles');
 
-function invalidateCache(mtimes, cache, done) {
+function invalidateCache(b, mtimes, cache, done) {
   assertExists(mtimes);
 
   invalidateModifiedFiles(mtimes, Object.keys(cache), function(file) {
     delete cache[file];
-  }, done);
+  }, function (file) {
+    return invalidateModifiedStream(b, cache, file)
+  }, function(err, invalidated, deleted) {
+    if (deferedQueue.length > 0) {
+      b.on('_ready', function () {
+        invalidated = invalidated.concat(deferedQueue.filter(function (file) {
+          return invalidateModifiedStream(b, cache, file)
+        }))
+        done(err, invalidated, deleted)
+      })
+
+      return
+    }
+
+    done(err, invalidated, deleted)
+  });
+}
+
+var deferedQueue = [];
```

Comment:
If multiple browserify runs are happening simultaneously (it doesn't happen when using browserify from the CLI but can be done when using browserify from node), then it looks like they'll all share the same deferedQueue variable which doesn't seem intended.